File: skills/statistical-data-manager/r/tests/testthat/test-core.R
source(file.path("..", "..", "R", "core.R"))

fixture <- file.path("..", "..", "..", "fixtures", "duplicate_rows.csv")

testthat::test_that("profile_data reports the shared fixture", {
  data <- read.csv(fixture, na.strings = "", stringsAsFactors = FALSE)
  result <- profile_data(data)

  testthat::expect_equal(result$row_count, 4)
  testthat::expect_equal(result$column_count, 3)
  testthat::expect_equal(result$columns, c("participant_id", "visit", "value"))
  testthat::expect_equal(unname(result$missing_count[["value"]]), 2)
  testthat::expect_equal(result$duplicate_row_count, 2)
})

testthat::test_that("inspect_data proposes review without mutating", {
  data <- read.csv(fixture, na.strings = "", stringsAsFactors = FALSE)
  before <- data
  result <- inspect_data(data)

  testthat::expect_identical(data, before)
  testthat::expect_length(result$findings, 1)
  testthat::expect_equal(result$findings[[1]]$code, "duplicate_rows")
  testthat::expect_equal(result$findings[[1]]$status, "needs_context")
})

File: skills/statistical-data-manager/python/tests/test_core.py
from pathlib import Path
import unittest

import pandas as pd

from suai_data_manager import inspect, profile


FIXTURE = Path(__file__).parents[2] / "fixtures" / "duplicate_rows.csv"


class CoreTests(unittest.TestCase):
    def test_profile_reports_shared_fixture(self) -> None:
        data = pd.read_csv(FIXTURE)
        result = profile(data)

        self.assertEqual(result["row_count"], 4)
        self.assertEqual(result["column_count"], 3)
        self.assertEqual(result["columns"], ["participant_id", "visit", "value"])
        self.assertEqual(result["missing_count"]["value"], 2)
        self.assertEqual(result["duplicate_row_count"], 2)

    def test_inspect_proposes_review_without_mutating(self) -> None:
        data = pd.read_csv(FIXTURE)
        before = data.copy(deep=True)

        result = inspect(data)

        pd.testing.assert_frame_equal(data, before)
        self.assertEqual(len(result.findings), 1)
        self.assertEqual(result.findings[0].code, "duplicate_rows")
        self.assertEqual(result.findings[0].status, "needs_context")


if __name__ == "__main__":
    unittest.main()
